src/player/player.c

```diff
@@ -563,7 +564,7 @@ void playerUpdateSounds(struct Player* player) {
         soundPlayerPlay(soundsConcreteFootstep[3], 1.0f, 0.5f, NULL, NULL, SoundTypeAll);
         player->flags &= ~PlayerJustJumped;
     }
-    if (flags & PlayerJustLanded) {
+    if (player->body.velocity.y > PLAYER_LAND_STEP_VEL && flags & PlayerJustLanded) {
```

Comment:
A few things:
* If the velocity check fails then the `PlayerJustLanded` flag will not be reset and the sound will play the next time the velocity check passes (which may be while jumping or in the air). Clearing the flag should be unconditional. But this will likely be irrelevant after fixing the next point.
* I tried this out and by the time execution gets here, the y velocity has already been zeroed out. You could move the check to `playerUpdateFooting()` to avoid that. It would also mean the `PlayerJustLanded` flag could be removed.
* Player velocity will be negative when landing. So the check should be something like this.
    ```c
    player->body.velocity.y < -PLAYER_LAND_STEP_VEL
    ```

Reply:
> by the time execution gets here, the y velocity has already been zeroed out

I had not realized this at the time. I don't think I had my debug environment working at the time. 😊 I flipped the sign as a sanity check and found that it "worked".

Anyway, I've implemented this correctly now.